File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int n = nums.length;

        int low = -1, mid = 0, high = n;
        while (mid < high) {
            if (nums[mid] == 0) {
                low++;
                nums[mid] = nums[low] ^ nums[mid] ^ (nums[low] = nums[mid]);
                mid++;
            } else if (nums[mid] == 1) {
                mid++;
            } else {
                high--;
                nums[mid] = nums[high] ^ nums[mid] ^ (nums[high] = nums[mid]);
            }
        }
    }
}

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        low, mid, high = -1, 0, len(nums)

        while (mid < high):
            if 0 == nums[mid]:
                low += 1
                nums[low], nums[mid] = nums[mid], nums[low]
                mid += 1
            elif 1 == nums[mid]:
                mid += 1
            else:
                high -= 1
                nums[mid], nums[high] = nums[high], nums[mid]
                
